test/tools/runner/config.ts

```diff
@@ -427,6 +439,63 @@ export class TestConfiguration {
   makeAtlasTestConfiguration(): AtlasTestConfiguration {
     return new AtlasTestConfiguration(this.uri, this.context);
   }
+
+  loggingEnabled = false;
+  logs = [];
+  /**
+   * Known flaky tests that we want to turn on logging for
+   * so that we can get a better idea of what is failing when it fails
+   */
+  testsToEnableLogging = [
+    'Client Side Encryption (Unified) namedKMS-rewrapManyDataKey rewrap to azure:name1',
+    'Server Discovery and Monitoring Prose Tests Connection Pool Management ensure monitors properly create and unpause connection pools when they discover servers',
+    'CSOT spec tests legacy timeouts behave correctly for retryable operations operation succeeds after one socket timeout - aggregate on collection'
+  ];
+
+  setupLogging(options: MongoClientOptions, s?: string) {
+    s ??= new ObjectId().toString();
+    this.logs = [];
+    const write = log => this.logs.push({ t: log.t, s, ...log });
+    options.mongodbLogPath = { write };
+    options.mongodbLogComponentSeverities = { default: 'trace' };
+    options.mongodbLogMaxDocumentLength = 300;
+    return options;
+  }
+
+  beforeEachLogging(ctx: Context) {
+    this.loggingEnabled = this.testsToEnableLogging.includes(ctx.currentTest.fullTitle());
+  }
+
+  afterEachLogging(ctx: Context) {
+    if (this.loggingEnabled && ctx.currentTest.state === 'failed') {
+      for (const log of this.logs) {
+        console.error(
+          JSON.stringify(
```

Comment:
Just use `EJSON.stringify` here?

Reply:
I can't replace values that aren't supported in EJSON: https://jira.mongodb.org/browse/NODE-4122

Reply:
What isn't EJSON stringifiyable from our logger?  I see no errors locally

Reply:
Set, Errors, and JSON removes undefined fields are a few examples of information loss that could be valuable to keep. I think I captured the cases in the replacer. Using util.inspect just catches all possible JS values and doesn't accidentally hide anything which is why I prefer it